core/src/packet.rs

```diff
@@ -65,7 +65,7 @@ impl fmt::Debug for Packet {
 impl Default for Packet {
     fn default() -> Packet {
         Packet {
-            data: [0u8; PACKET_DATA_SIZE],
+            data: unsafe { std::mem::uninitialized() },
```

Comment:
Is it possible this opens a buffer overrun attack via serde deserialize?

Reply:
I don't see how this is any better or worse than an application sending 512 bytes of garbage, only the first 220 bytes of which look like a signed transaction...  we only pay attention to bytes on the wire, and we read indiscriminately from the network

Reply:
our recv_from could put a null after the last received byte if serde is so borken that nulls are somehow special